File: solutions/231.Power_of_Two/AC_bitwize_1.cpp
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_bitwize_1.cpp
*  Create Date: 2015-07-12 19:47:54
*  Descripton:   
*/

#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool isPowerOfTwo(int n) {
        for (int i = 0; i < 32; ++i)
            if (((n >> i) == 1) && (1 << i) == n)
                return true;
        return false;
    }
};

int main() {
    return 0;
}

File: solutions/231.Power_of_Two/AC_bitwize_1.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# Author:      illuz <iilluzen[at]gmail.com>
# File:        AC_bitwize_1.py
# Create Date: 2015-07-12 20:02:44
# Usage:       AC_bitwize_1.py 
# Descripton:  

class Solution:
    def isPowerOfTwo(self, n):
        return (n > 0) and (n & (n-1)) == 0
